### PR TITLE
bluewalker: init at 0.3.0

### PR DESCRIPTION
**Pull request by cimm**
###### Description of changes

I packaged [bluewalker](https://gitlab.com/jtaimisto/bluewalker), a command line Bluetooth LE scanner written in go since I would like to keep using it after migrating form Fedora. Might as well contribute it back to the main repository if anyone else wants to use it.

This is the first time I package a Nix application so maybe someone can tell me if I am missing something.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/bluetooth/bluewalker/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-spuJRiNiaBV4EsetUq8vUfR6ejUNZxLhVzS3AZZyrrQ=";
+  };
+
+  vendorSha256 = "189qs6vmx63vwsjmc4qgf1y8xjsi7x6l1f5c3kd8j8jnagl26z4h";
+
+  ldflags = [ # omit some things for a smaller binary
+    "-w" # omit the DWARF symbol table
+    "-s" # omit symbol table and debug info
+  ];
```
Comment (fabaff) [inline pkgs/tools/bluetooth/bluewalker/default.nix:19]: Most Go developers and package maintainers are familiar with those flags.

```suggestion
  ldflags = [
    "-w"
    "-s"
  ];
```

File: pkgs/tools/bluetooth/bluewalker/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "Simple command line Bluetooth LE scanner";
+    homepage = src.meta.homepage;
```
Comment (fabaff) [inline pkgs/tools/bluetooth/bluewalker/default.nix:23]: Much easier for the reviews who are processing update and alike if it's possible to go to upstream's website which additional effort.

```suggestion
    homepage = "https://gitlab.com/jtaimisto/bluewalker";
```

File: pkgs/tools/bluetooth/bluewalker/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    changelog = "https://gitlab.com/jtaimisto/${pname}/-/raw/master/CHANGELOG";
```
Comment (fabaff) [inline pkgs/tools/bluetooth/bluewalker/default.nix:24]: This URL points to `master`. 

```suggestion
    changelog = "https://gitlab.com/jtaimisto/bluewalker/-/tags/v${version}";
```

or 

```suggestion
    changelog = "https://gitlab.com/jtaimisto/bluewalker/-/blob/v${version}/CHANGELOG";
```

**Comment by cimm** [comment]
Great feedback @fabaff. Makes a lot of sense and thanks for taking the time helping a newbie!

**Comment by fabaff** [comment]
Related to https://github.com/NixOS/nixpkgs/issues/81418

**Pull request merged**